subiquity/controller.py

```diff
@@ -26,6 +26,45 @@
 
 class SubiquityController(BaseController):
 
+    autoinstall_key = None
+    autoinstall_default = None
+
+    def __init__(self, app):
+        super().__init__(app)
+        self.autoinstall_applied = False
+        if app.autoinstall_config:
+            self.load_autoinstall_data(
+                app.autoinstall_config.get(
+                    self.autoinstall_key,
+                    self.autoinstall_default))
```

Comment:
Pre-populating the autoinstall_config dictionary with keys which may be None (due to the default value being set) means that checking for presence of a key w.r.t to whether subiquity uses a default value or the user provided can be difficult.

Reply:
Yes, there are a few different ways of looking at this. This API seems reasonably ok. load_install_data has access to self.app.autoinstall_config if it needs it...